pypdf/_writer.py

```diff
@@ -829,18 +829,19 @@ def update_page_form_field_values(
                     if writer_annot.get(FieldDictionaryAttributes.FT) == "/Btn":
                         writer_annot.update(
                             {
-                                NameObject(
-                                    AnnotationDictionaryAttributes.AS
-                                ): NameObject(fields[field])
+                                NameObject(FieldDictionaryAttributes.V): NameObject(fields[field]),
```

Comment:
Note for myself:

> The V entry in the field dictionary (see Table 220) holds a name object representing the check box’s
appearance state, which shall be used to select the appropriate appearance from the appearance dictionary.

Reply:
> (Optional; inheritable) The field’s value, whose format varies
depending on the field type. See the descriptions of individual field
types for further information.